editors/emacs/racer.el

```diff
@@ -296,27 +184,30 @@ foo(bar, |baz); -> foo|(bar, baz);"
     (-when-let (rust-sym (symbol-at-point))
       (-some->>
        ;; then look at the current completion possiblities,
-       (racer--candidates)
+       (racer-complete)
        ;; extract the possibility that matches this symbol exactly
-       (--filter (equal it (symbol-name rust-sym)))
+       (--filter (string= it (symbol-name rust-sym)))
        (-first-item)
        ;; and return the prototype that Racer gave us.
-       (get-text-property 0 'contextstr)
+       (get-text-property 0 'ctx)
        ;; Finally, apply syntax highlighting for the minibuffer.
        (racer--syntax-highlight)))))
 
-;;;###autoload
-(defun racer-turn-on-eldoc ()
-  "Enable eldoc using Racer."
-  (make-local-variable 'eldoc-documentation-function)
-  (setq-local eldoc-documentation-function #'racer--eldoc)
-  (eldoc-mode))
-
-;;;###autoload
-(defun racer-turn-off-eldoc ()
-  "Disable eldoc using Racer."
-  (kill-local-variable 'eldoc-documentation-function)
-  (eldoc-mode -1))
+(defvar racer-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "M-.") #'racer-find-definition)
+    map))
+
+(define-minor-mode racer-mode
+  "Minor mode for racer."
+  :lighter " racer"
+  :keymap racer-mode-map
+  (setq-local eldoc-documentation-function #'racer-eldoc)
+  (make-local-variable 'completion-at-point-functions)
+  (add-to-list 'completion-at-point-functions #'racer-complete-at-point))
+
+(define-obsolete-function-alias 'racer-turn-on-eldoc 'eldoc-mode)
```

Comment:
Shouldn't this be `(define-obsolete-function-alias 'racer-turn-on-eldoc 'racer--eldoc)`?